### ISSUE TITLE
Springfox Swagger generating requests with port 80 for HTTPS URLS

### ISSUE THREAD

**Issue by martin3361** [springfox/springfox#1634]
As raised here: http://stackoverflow.com/questions/33117478/springfox-swagger-generating-requests-with-port-80-for-https-urls

as /v2/api-docs says, there is a property host and it does include port. How is expected proper replacement with 443 when necessary?

We have deployment of Swagger on SSL enabled endpoint, and URLs requested by Swagger look like:
https://host:80/service/endpoint?param=1
this messes up everything.

**Comment by dilipkrish** [comment]
You can override host and port in the docket. Please use v2.6.1+

**Comment by martin3361** [comment]
we don't want to specify hosts and ports because we have dynamic hosts and ports in AWS + microservice architecture. This is no solution for us.
We tried to upgrade to newer version, but had other problems, which I cannot recall right now.

**Comment by aug70** [comment]
springfox.documentation.swagger.v2.host property can only override host and port. But what about protocol? This makes it still unusable in development environment.

**Comment by dilipkrish** [comment]
This is fixed in 2.7.0-SNAPSHOT. Would you mind trying it out and seeing if it fits the bill?

**Comment by matdelong-wb** [comment]
@dilipkrish : Could you tell me when version 2.7.0 will be released? Thanks!

**Comment by zhudan** [comment]
same problem！

**Comment by flavioeb-ciandt** [comment]
I updated to version 2.7.0 and it didn't fix the problem. Was the issue closed? I have same scenario as @martin3361

**Comment by Divzz** [comment]
Is there a provision to set the port alone and not the "hostname:port" everytime?

**Comment by dilipkrish** [comment]
Port is always inferred or overrides using the `Docket#host("some-host:some_port)` api.

**Comment by Divzz** [comment]
I have version 2.7.0. I have deployed my application on kubernetes and accessing it as a https URL. The Swagger UI comes up fine but when I press "Try it out!" it forwards the request to port 80 although I havent defined it anywhere. I saw in v2/api-docs that the host is getting appended with port 80 and messes up the request. So my question is shouldn't port 443 be inferred when it knows its a https URL. Should I be overriding it using Docket#host(hostname:port)? Or is there something else that I need to set to ensure Swagger UI infers port 443?
